telemetry.py

```diff
@@ -0,0 +1,158 @@
+import os
+import time
+from datetime import datetime, UTC
+import threading
+from queue import Queue
+
+from prometheus_client import start_http_server, Gauge, Enum, Counter, Histogram
+import psutil
+
+from mission import Telemetry
+
+
+class KSPTelemetry:
+    """
+    A class to handle publishing metrics periodically
+    """
+
+    def __init__(self):
+        self.publish_rate_hz = int(os.environ.get('TELEM_PUBLISH_RATE', 0))
+
+        # all the metrics which come from krpc stream and mission.Telemetry class
+        self.surface_altitude = Gauge('surface_altitude', 'surface_altitude in meters')
+        self.altitude = Gauge('altitude', 'altitude in meters')
+        self.apoapsis = Gauge('apoapsis', 'apoapsis in meters')
+        self.velocity = Gauge('velocity', 'velocity in meters per second')
+        self.vertical_vel = Gauge('vertical_vel', 'vertical_vel in meters per second')
+        self.horizontal_vel = Gauge('horizontal_vel', 'horizontal_vel in meters per second')
+        self.periapsis = Gauge('periapsis', 'periapsis in meters')
+
+        # host metrics
+        self.system_cpu_percent = Gauge('system_cpu_percent', 'total host CPU usage in percent')
+
+        # metrics which come from control system calculations, not krpc streams
+        self.gauge_metrics = {}
+        self.counter_metrics = {}
+        self.histogram_metrics = {}
+        self.enum_metrics = {}
+
+        self.gnc_debug = os.environ.get('GNC_DEBUG', None) is not None
+
+
+
+    def start_metrics_server(self, port: int = 8012):
+        # todo some kind of lock or better error handling here if this class is instantiated > once
+        start_http_server(port)
+
+
+    def start_telem_publish(self, telemetry: Telemetry) -> bool:
+        if self.publish_rate_hz == 0:
+            print('not starting telemetry publish thread')
+            return False
+
+        updater_thread = threading.Thread(target=self._telem_publish_thread, args=(telemetry,), daemon=True)
+        updater_thread.start()
+        return True
+
+
+    def _telem_publish_thread(self, telemetry: Telemetry):
+        while True:
+            self.surface_altitude.set(telemetry.surface_altitude())
+            self.altitude.set(telemetry.altitude())
+            self.apoapsis.set(telemetry.apoapsis())
+            self.velocity.set(telemetry.velocity())
+            self.vertical_vel.set(telemetry.vertical_vel())
+            self.horizontal_vel.set(telemetry.horizontal_vel())
+            self.periapsis.set(telemetry.periapsis())
+
+            self.system_cpu_percent.set(psutil.cpu_percent())
+
+            time.sleep(1/self.publish_rate_hz)
+
+    # -------------------------------------------------------------
+    # Worker Thread Function
+    # -------------------------------------------------------------
+    def metric_publisher(self, queue: Queue):
```

Comment:
for consistency with other threaded thing in this class, this fn should be internal and called by a fn like:

```suggestion

    def start_metrics_queue_thread(self, queue: Queue) -> bool:
        updater_thread = threading.Thread(target=self._metric_publisher_thread, args=(queue,), daemon=True)
        updater_thread.start()
        return True
    
      def _metric_publisher_thread(self, queue: Queue):
```

this would be nice for consumers b/c it would cut down on work of needing to start the thread